packages/expo-sqlite/ios/SQLiteModule.swift

```diff
@@ -51,8 +64,17 @@ public final class SQLiteModule: Module {
       }
     }
 
+    OnStartObserving {
+      hasListeners = true
+    }
+
+    OnStopObserving {
+      hasListeners = false
+    }
+
     OnDestroy {
       cachedDatabases.values.forEach {
+        executeSql(sql: "SELECT crsql_finalize()", with: [], for: $0, readOnly: false)
```

Comment:
👍

Might be worth pushing this sort of detail down into the SQLite bindings themselves so users can never forget to `finalize` the extension.

fwiw -- the libsql fork of sqlite has fixed the need to do this: https://github.com/libsql/libsql/issues/62